addOns/encoder/src/main/java/org/zaproxy/addon/encoder/processors/script/EncodeDecodeScriptHelper.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2022 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.encoder.processors.script;
+
+import org.apache.commons.lang3.StringUtils;
+import org.parosproxy.paros.Constant;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeProcessor;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeProcessorItem;
+import org.zaproxy.addon.encoder.processors.EncodeDecodeProcessors;
+import org.zaproxy.addon.encoder.processors.predefined.DefaultEncodeDecodeProcessor;
+
+public class EncodeDecodeScriptHelper {
+
+    public EncodeDecodeScriptHelper() {
+        // Nothing to do
+    }
+
+    /**
+     * Returns the processor with the given identified, or a default processor which will state the
+     * sought one was not found.
+     *
+     * @param id of the processor to be returned.
+     * @return the {@Code EncodeDecodeProcessor} which was requested, or a default processor.
+     */
+    public EncodeDecodeProcessor getProcessorById(String id) {
+        for (EncodeDecodeProcessorItem item : EncodeDecodeProcessors.getPredefinedProcessors()) {
+            if (item.getId().equalsIgnoreCase(EncodeDecodeProcessors.PREDEFINED_PREFIX + id)) {
+                return item.getProcessor();
+            }
+        }
+        return getDefaultProcessor(id);
+    }
+
+    /**
+     * Returns the first predefined processor with an id that matches based on a string contains
+     * condition ignoring case, or a default processor which will state the sought one was not
+     * found. Ex: "md5" would match "md5hash" or "urle" would match "urlencode".
+     *
+     * @param id of the processor to be returned.
+     * @return the first {@Code EncodeDecodeProcessor} matched, or a default processor.
+     */
+    public EncodeDecodeProcessor getProcessorByFuzzyMatch(String id) {
```

Comment:
What's the use case for this one? This is not that reliable, tomorrow we might add a processor that overlaps an existing name and the script would get a different processor than expected.

Reply:
I guess. I just figured there isn't always a good way for users to know what the IDs are.

Reply:
We can link to the existing code which has the names or explicitly expose the predefined processors through a method (e.g. `getMd5Hasher()`), either way if we publish the JavaDoc the user can check the IDs or the methods there.